### PR TITLE
Hpc best practices

### PR DESCRIPTION
**Pull request by b10n1k**
- Use of Mojo::Base https://docs.mojolicious.org/Mojo/Base
      strict and warnings are inherited by this Class
- Use kernel maillist for maintainer                                                                                                                                                                                                 
                                                                                                                                                                                                                                             
With this PR all the HPC modules introduce the `signatures` flag which is                                                                                                                                                                    
supported by `Mojo::Base`. The signature declares lexical variables that are                                                                                                                                                                 
in scope for the block                                                                                                                                                                                                                       
and every subroutine invocation, first is checked for its signature.


- Related ticket: https://progress.opensuse.org/issues/xyz
- Verification run: 
https://openqa.suse.de/tests/8266698
https://openqa.suse.de/tests/8266713#dependencies

### CODE REVIEW

File: lib/hpc/cluster.pm
Comment (okurz) [inline lib/hpc/cluster.pm:3]: don't update copyright years with ranges, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md?plain=1#L49

File: tests/hpc/ganglia_client.pm
Comment (okurz) [inline tests/hpc/ganglia_client.pm:9]: I suggest to keep that. Even if fate is not actively used anymore it still acts as a reference explaining why we have test modules at all

**Review by okurz — CHANGES_REQUESTED** [review]

File: lib/hpc/cluster.pm
Comment (b10n1k) [reply lib/hpc/cluster.pm]: *this change is ready for review*

File: tests/hpc/ganglia_client.pm
Comment (b10n1k) [reply tests/hpc/ganglia_client.pm:9]: *this change is ready for review*

**Review by okurz — APPROVED** [review]
Please await review from former and updated test module maintainers

**Comment by schlad** [comment]
nice cleanup! thx @b10n1k

**Comment by dzedro** [comment]
https://openqa.suse.de/tests/8284597

**Pull request merged**
